src/mjolnir/osmdata.cc

```diff
@@ -308,6 +320,31 @@ bool write_linguistic(const std::string& filename, const LinguisticMultiMap& lin
   return true;
 }
 
+bool write_conditional_speed_limits(const std::string& filename,
+                                    const ConditionalSpeedLimitsMultiMap& speed_map) {
+  // Open file and truncate
+  std::ofstream file(filename.c_str(), std::ios::out | std::ios::binary | std::ios::trunc);
+  if (!file.is_open()) {
+    LOG_ERROR("write_linguistic failed to open output file: " + filename);
+    return false;
+  }
+
+  // Convert the multi map into a flat representation
+  std::vector<TempConditionalSpeedLimit> flat;
+  flat.reserve(speed_map.size());
+  for (auto it = speed_map.cbegin(); it != speed_map.cend(); ++it) {
+    flat.emplace_back(it->first, it->second);
```

Comment:
Can you delete `TmpConditionalSpeedLimit` and just write to file in the loop?

Reply:
Well, all items are stored via temp flat structure and via flat vector. I definitely don't want to refactor whole `osmdata.cc` in this PR.

Reply:
Addressed in b08327ce2b9c8cbbbf0ac0bd8826a39f096baa17